src/Catalyst.Node.POA.CE.Tests/IntegrationTests/PoaConsensusTests.cs

```diff
@@ -0,0 +1,115 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Autofac;
+using Catalyst.Core.Modules.Consensus.Cycle;
+using Catalyst.Core.Modules.Cryptography.BulletProofs;
+using Catalyst.TestUtils;
+using FluentAssertions;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Catalyst.Node.POA.CE.Tests.IntegrationTests
+{
+    public sealed class PoaConsensusTests : FileSystemBasedTest
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Implement 'IDisposable' in this class and use the 'Dispose' method to call 'Dispose' on '_endOfTestCancellationSource'.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4594394)